### PR TITLE
Fixed some errors and rename

### PR DESCRIPTION
**Pull request by iattempt**
**Fixes issue:**

#3274 

**Changes:**

1. fixed coding style
2. separate src/test
3. fixed logic error
4. rename files/directory

### CODE REVIEW

File: code/unclassified/src/spiral_print/spiral_print.cpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+			res << vec[begRow][i] << " ";
+		++begRow;
+
+		// Print the end col
+		for (int i = begRow; i <= endRow; ++i)
+			res << vec[i][endCol] << " ";
+		--endCol;
+
+		// Print the end row
+		if (endRow >= begRow)
```
Comment (iattempt) [inline code/unclassified/src/spiral_print/spiral_print.cpp]: fix error result while even rows/cols gave.

File: code/unclassified/test/spiral_printing/test_spiral_print.cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	   2
+	   3
+	 */
+	vec.clear();
+	vec.push_back({1});
+	assert(spiralPrint(vec, vec.size(), vec[0].size()) == "1");
+	vec.push_back({2});
+	assert(spiralPrint(vec, vec.size(), vec[0].size()) == "1 2");
+	vec.push_back({3});
+	assert(spiralPrint(vec, vec.size(), vec[0].size()) == "1 2 3");
```
Comment (iattempt) [inline code/unclassified/test/spiral_printing/test_spiral_print.cpp]: TODO: use random texts/cols/rows for tests.

File: code/unclassified/src/spiral_printing/spiral_print.cpp [hunk 3]
```diff
@@ -1,64 +1,47 @@
...
-        for(int j=0; j<n; j++)
-            cin >> a[i][j];
-    spiralPrint(a,m,n);
-
-
-
-
-
-return 0;
+#include <vector>
```
Comment (iattempt) [inline code/unclassified/src/spiral_printing/spiral_print.cpp]: change to standard header.

File: code/unclassified/src/spiral_printing/spiral_print.cpp [hunk 4]
```diff
@@ -1,64 +1,47 @@
...
+#include <sstream>
+
+std::string spiralPrint(std::vector<std::vector<int>> vec, int row, int col)
```
Comment (iattempt) [inline code/unclassified/src/spiral_printing/spiral_print.cpp]: rename variables for readable.

File: code/unclassified/src/spiral_printing/spiral_print.cpp [hunk 5]
```diff
@@ -1,64 +1,47 @@
...
+				res << vec[i][begCol] << " ";
+			++begCol;
+		}
+	}
+
+	auto s = res.str();
+	if (!s.empty())
+		s.pop_back();
+
+	return s;
```
Comment (iattempt) [inline code/unclassified/src/spiral_printing/spiral_print.cpp]: more useful, now the function can be used in that append to string or direct print.
TODO: to be class

File: code/unclassified/src/spiral_printing/spiral_print.cpp [hunk 6]
```diff
@@ -1,64 +1,47 @@
...
+			res << vec[begRow][i] << " ";
+		++begRow;
+
+		// Print the end col
+		for (int i = begRow; i <= endRow; ++i)
+			res << vec[i][endCol] << " ";
+		--endCol;
+
+		// Print the end row
+		if (endRow >= begRow)
```
Comment (iattempt) [inline code/unclassified/src/spiral_printing/spiral_print.cpp]: fix error result while even rows/cols gave.

File: code/unclassified/src/spiral_printing/spiral_print.cpp [hunk 7]
```diff
@@ -1,64 +1,47 @@
...
-    spiralPrint(a,m,n);
-
-
-
-
-
-return 0;
+#include <vector>
+#include <sstream>
+
```
Comment (iattempt) [inline code/unclassified/src/spiral_printing/spiral_print.cpp]: removed using directive in global.

File: code/unclassified/src/spiral_print/spiral_print.cpp [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+#include <vector>
+#include <sstream>
+
+std::string spiralPrint(std::vector<std::vector<int>> vec, int row, int col)
+{
+	int begRow = 0,
+	    endRow = row - 1,
+	    begCol = 0,
+	    endCol = col - 1;
+	std::ostringstream res{};
```
Comment (arnavb) [inline code/unclassified/src/spiral_print/spiral_print.cpp]: No need for the braces.

File: code/unclassified/test/spiral_printing/test_spiral_print.cpp [hunk 9]
```diff
@@ -0,0 +1,54 @@
+#include "../../src/spiral_printing/spiral_print.cpp"
+#include <cassert>
+
+int main()
+{
+	using namespace std;
+	vector<vector<int>> vec{};
```
Comment (arnavb) [inline code/unclassified/test/spiral_printing/test_spiral_print.cpp]: Don't need braces

**Comment by iattempt** [comment]
@OpenGenus/cosmos-maintainers Please review.

File: code/unclassified/src/spiral_print/spiral_print.cpp (hunk 8 above)
Comment (arnavb) [inline code/unclassified/src/spiral_print/spiral_print.cpp]: No need for braces here.

File: code/unclassified/src/spiral_print/spiral_print.cpp [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+		if (begCol <= endCol)
+		{
+			for (int i = endRow; i >= begRow; --i)
+				res << vec[i][begCol] << " ";
+			++begCol;
+		}
+	}
+
+	auto s = res.str();
+	if (!s.empty())
```
Comment (arnavb) [inline code/unclassified/src/spiral_print/spiral_print.cpp]: https://github.com/OpenGenus/cosmos/blob/master/guides/coding_style/c++/README.md#indentations

> Use only spaces, and indent 4 spaces at a time.
We use spaces for indentation. Do not use tabs in our code.

(I am quoting you 🤣 )

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**